File: 0728-self-dividing-numbers/0728-self-dividing-numbers.py
class Solution(object):
    def selfDividingNumbers(self, left, right):
        """
        :type left: int
        :type right: int
        :rtype: List[int]
        """
        ans = []
        for i in range(left,right+1):
            if i<10:
                ans.append(i)
            elif i%10==0:
                continue
            else:
                tem=str(i)
                tem2=True
                for ch in tem:
                    if int(ch)==0:
                        tem2 = False
                        break
                    elif i%int(ch)!=0:
                        tem2 = False
                        break
                if tem2:
                    ans.append(i)
        return ans
        

File: 0728-self-dividing-numbers/0728-self-dividing-numbers.ts
function selfDividingNumbers(left: number, right: number): number[] {
    let ans = []
    for(let i=left;i<=right;i++){
        if(i<10){
            ans.push(i)
        } else if(i%10==0){
            continue
        } else {
            let tem = String(i)
            let tem2 = true
            for(let j=0;j<tem.length;j++){
                let t = Number(tem[j])
                if(tem[j]=="0" || i%t!=0){
                    tem2 = false
                    break
                }
            }
            if(tem2){
                ans.push(i)
            }
        }
    }
    return ans
};